File: demos/jans-chip/android/app/src/main/java/io/jans/chip/modal/OPConfiguration.java
package io.jans.chip.modal;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

import io.jans.chip.modal.OperationError;
@Entity(tableName = "OP_CONFIGURATION")
public class OPConfiguration {
    public OPConfiguration() {
    }

    public OPConfiguration(String sno,
                           String issuer,
                           String registrationEndpoint,
                           String tokenEndpoint,
                           String userinfoEndpoint,
                           String authorizationChallengeEndpoint,
                           String revocationEndpoint,
                           boolean isSuccessful) {
        this.sno = sno;
        this.issuer = issuer;
        this.registrationEndpoint = registrationEndpoint;
        this.tokenEndpoint = tokenEndpoint;
        this.userinfoEndpoint = userinfoEndpoint;
        this.authorizationChallengeEndpoint = authorizationChallengeEndpoint;
        this.revocationEndpoint = revocationEndpoint;
        this.isSuccessful = isSuccessful;
    }

    public OPConfiguration(boolean isSuccessful, OperationError operationError) {
        this.isSuccessful = isSuccessful;
        this.operationError = operationError;
    }
    @NonNull
    @PrimaryKey
    @SerializedName("SNO")
    private String sno;
    @ColumnInfo(name = "ISSUER")
    @SerializedName("issuer")
    private String issuer;
    @ColumnInfo(name = "REGISTRATION_ENDPOINT")
    @SerializedName("registration_endpoint")
    private String registrationEndpoint;
    @ColumnInfo(name = "TOKEN_ENDPOINT")
    @SerializedName("token_endpoint")
    private String tokenEndpoint;
    @ColumnInfo(name = "USERINFO_ENDPOINT")
    @SerializedName("userinfo_endpoint")
    private String userinfoEndpoint;
    @ColumnInfo(name = "AUTHORIZATION_CHALLENGE_ENDPOINT")
    @SerializedName("authorization_challenge_endpoint")
    private String authorizationChallengeEndpoint;
    @ColumnInfo(name = "REVOCATION_ENDPOINT")
    @SerializedName("revocation_endpoint")
    private String revocationEndpoint;
    @Ignore
    private boolean isSuccessful;
    @Ignore
    private OperationError operationError;
    public String getRevocationEndpoint() {
        return revocationEndpoint;
    }

    public void setRevocationEndpoint(String revocationEndpoint) {
        this.revocationEndpoint = revocationEndpoint;
    }

    public String getIssuer() {
        return issuer;
    }

    public void setIssuer(String issuer) {
        this.issuer = issuer;
    }

    public String getAuthorizationChallengeEndpoint() {
        return authorizationChallengeEndpoint;
    }
    public void setAuthorizationChallengeEndpoint(String authorizationChallengeEndpoint) {
        this.authorizationChallengeEndpoint = authorizationChallengeEndpoint;
    }

    public String getRegistrationEndpoint() {
        return registrationEndpoint;
    }

    public void setRegistrationEndpoint(String registrationEndpoint) {
        this.registrationEndpoint = registrationEndpoint;
    }

    public String getTokenEndpoint() {
        return tokenEndpoint;
    }

    public void setTokenEndpoint(String tokenEndpoint) {
        this.tokenEndpoint = tokenEndpoint;
    }

    public String getUserinfoEndpoint() {
        return userinfoEndpoint;
    }

    public void setUserinfoEndpoint(String userinfoEndpoint) {
        this.userinfoEndpoint = userinfoEndpoint;
    }
    public boolean isSuccessful() {
        return isSuccessful;
    }

    public void setSuccessful(boolean successful) {
        isSuccessful = successful;
    }

    public OperationError getOperationError() {
        return operationError;
    }

    public void setOperationError(OperationError operationError) {
        this.operationError = operationError;
    }

    @NonNull
    public String getSno() {
        return sno;
    }

    public void setSno(@NonNull String sno) {
        this.sno = sno;
    }

    @Override
    public String toString() {
        return "OPConfiguration{" +
                "issuer='" + issuer + '\'' +
                ", registrationEndpoint='" + registrationEndpoint + '\'' +
                ", tokenEndpoint='" + tokenEndpoint + '\'' +
                ", userinfoEndpoint='" + userinfoEndpoint + '\'' +
                ", authorizationChallengeEndpoint='" + authorizationChallengeEndpoint + '\'' +
                '}';
    }
}

File: demos/jans-chip/android/app/src/main/java/io/jans/chip/model/OPConfiguration.kt
package io.jans.chip.model

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "OP_CONFIGURATION")
data class OPConfiguration(
    @NonNull
    @PrimaryKey(autoGenerate = false)
    @SerializedName("SNO")
    var sno: String,
    @ColumnInfo(name = "ISSUER")
    @SerializedName("issuer")
    var issuer: String?,
    @ColumnInfo(name = "REGISTRATION_ENDPOINT")
    @SerializedName("registration_endpoint")
    var registrationEndpoint: String?,
    @ColumnInfo(name = "TOKEN_ENDPOINT")
    @SerializedName("token_endpoint")
    var tokenEndpoint: String?,
    @ColumnInfo(name = "USERINFO_ENDPOINT")
    @SerializedName("userinfo_endpoint")
    var userinfoEndpoint: String?,
    @ColumnInfo(name = "AUTHORIZATION_CHALLENGE_ENDPOINT")
    @SerializedName("authorization_challenge_endpoint")
    var authorizationChallengeEndpoint: String?,
    @ColumnInfo(name = "REVOCATION_ENDPOINT")
    @SerializedName("revocation_endpoint")
    var revocationEndpoint: String?,
    @ColumnInfo(name = "BIOMETRIC_ENROLLED")
    @SerializedName("biometric_enrolled")
    var biometricEnrolled: Boolean = false,

) {
    @Ignore
    var isSuccessful: Boolean = false

    @Ignore
    var errorMessage: String = ""

    @ColumnInfo(name = "FIDO_URL")
    var fidoUrl: String? = null
}